client/extensions/woocommerce/woocommerce-services/lib/utils/get-address-values.js

```diff
@@ -0,0 +1,25 @@
+/** @format */
+
+/**
+ * External dependencies
+ */
+import { isPlainObject } from 'lodash';
+
+/**
+ * @param {object} addressData Object representing the current address UI state.
+ * - {object} addressData.values The address values, as entered by the user.
+ * - {object} addressData.normalized The address values, if any, as returned by the address verification endpoint.
+ * - {boolean} addressData.isNormalized True if the address has been normalized, false otherwise.
+ * - {boolean} addressData.selectNormalized True if the user has chosen to accept the normalized address, false otherwise.
+ * @returns {object} Object with the selected address values (postcode, country, state, etc), or an empty object on error.
```

Comment:
just curious, but why not `null` on error?

Reply:
That should never happen because in all the usages of this function, the `addressData` object has valid properties. But, in case it doesn't for some reason, returning an empty object means that code like this:
```js
const { postcode, state, country } = getAddressValues( addressData );
```
Would work just fine. It would render `undefined` or empty strings in some places in the UI, but that's better than crashing the whole page.
I don't know where you stand on the "crash spectacularly on error" vs "just do your best and keep going" approach to error handling, I guess that's a matter of personal preference :)

Reply:
thanks for the response. I take it subjectively as needs present themselves. in a case like this it can make sense. we just want to be really careful that we don't propagate an error deeper.

> just do your best and keep going

in my personal preference this is one of the worst things we can do both as developers and as teammates because it takes a knowable failure point and hides it deeper into the application where it will impact someone else unexpectedly and for seemingly no reason.

that is, having an empty `postcode` may be normal from a semantics viewpoint while assuming data is there that isn't in another situation can be disastrous. in any case we could always test for `{}` though that's why I asked initially for the `null`. `null` is nice because it sticks out immediately.

Reply:
> Having an empty postcode may be normal from a semantics viewpoint while assuming data is there that isn't in another situation can be disastrous.

It's not disastrous in this case. It would just populate empty data into a form or a string.

> In any case we could always test for {} though that's why I asked initially for the null.

Yeah, totally. If I were to check for that error condition in all the calls to this function, returning `null` would have been my first choice. I just don't think it's worth it in this case :)